### DISCUSSION TITLE
Installing service and web workers

### DISCUSSION THREAD

**Discussion by craigbilner** [freshframework/fresh#944] [Q&A]
What's the best practice for installing workers?

For my webworker the only way I could get it to work was to create in a `useEffect` hook using a path to root and serving a `.js` file from `static`. Ideally I'd like to use `.ts` but the auto created `Content-Type` is incorrect.

**Comment by Loshido** [comment]
You can use the bundle functionality like this

*on startup or using task start*
```ts
const command = new Deno.Command("deno", {
    args: [
        "bundle",
        "--platform=browser", // in case of web worker
        "worker.ts", // input
        "-o",
        "worker.js" // output
    ],
});

const { code, stdout, stderr } = await command.output();

if(code !== 0) {
    console.warn(new TextDecoder().decode(stderr))
} else {
    console.info(new TextDecoder().decode(stdout));
}
```
